src/collections/infra/repositories/CollectionsRepository.ts

```diff
@@ -40,11 +53,25 @@ export class CollectionsRepository extends ApiRepository implements ICollections
       })
     )
 
+    const inputLevelsRequestBody: NewCollectionInputLevelRequestPayload[] = []
+    collectionDTO.inputLevels.forEach((element) => {
```

Comment:
Maybe this could be done with functional programming as you also made with the `dataverseContacts` field.
Something like this:
```
const inputLevelsRequestBody: NewCollectionInputLevelRequestPayload[] =
  collectionDTO.inputLevels.map((inputLevel) => ({
    datasetFieldTypeName: inputLevel.datasetFieldName,
    include: inputLevel.include,
    required: inputLevel.required
  }))
```